### DISCUSSION TITLE
Laravel Telescope - using File as Storage Driver

### DISCUSSION THREAD

**Discussion by Sarthakj02** [laravel/framework#47265] [Q&A] [answered]
i want to store telescope entries in file instead of database but i am unable to figure it out can anyone help? is it even possible to use file as storage driver.
code from telescope.php

    'driver' => env('TELESCOPE_DRIVER', 'file'),

    'storage' => [
        'file' => [
            'path'       => storage_path('telescope'),
            'connection' => null,
        ],
    ],

**Comment by AnaelBonnafous** [comment, marked as answer]
I don't know if there is any other driver for telescope but what you can do is create a dedicated database for telescope. It will act as an unique file to store your telescope entries and they will not be present in your main database.

For that :

1. Create a telescope.sqlite file under database folder

2. In config/database.php, create a new connection with sqlite driver
```php
    'connections' => [

        'telescope_sqlite' => [
            'driver' => 'sqlite',
            'url' => env('TELESCOPE_DATABASE_URL'),
            'database' => env('TELESCOPE_DB_DATABASE', database_path('telescope.sqlite')),
            'prefix' => '',
            'foreign_key_constraints' => env('TELESCOPE_DB_FOREIGN_KEYS', true),
        ],

        // ...
    ],
```

3. In config/telescope.php, use this connection to store telescope entries
```php
    'driver' => env('TELESCOPE_DRIVER', 'database'),

    'storage' => [
        'database' => [
            'connection' => env('TELESCOPE_DB_CONNECTION', 'telescope_sqlite'),
            'chunk' => 1000,
        ],
    ],
```

- You can also set .env variables depending on your environment.

Hope it will help you!

**Reply by Sarthakj02** [reply]
Thank you
